runzi/execute/openquake/oq_hazard_task.py

```diff
@@ -176,6 +176,97 @@ def _sterilize_task_arguments_gmcmlt(self, ta):
+            self.model.hazard_config.set_sites(locations, backarc=backarc_flags)
+        else:
+            self.model.hazard_config.set_sites(locations, vs30=vs30s, backarc=backarc_flags)
+
+    def set_hazard_curve_parameters(self, task_arguments: Dict[str, Any]):
+        """Set hazard curve for hazard curve calculation
+
+        Args:
+            task_arguments: a dict of task arguments
+        """
+        imts = task_arguments["hazard_curve"]["imts"]
+        imtls = task_arguments["hazard_curve"]["imtls"]
+        self.model.hazard_config.set_iml(imts, imtls)
+
+    def set_disagg_matrix_parameters(self, task_arguments: Dict[str, Any]):
+        """Set disagg matrix coordinates and point on hazard curve to disaggregate
+
+        Args:
+            task_arguments: a dict of task arguments
+        """
+        self.model.hazard_config.set_iml_disagg(
+            imt=task_arguments["hazard_curve"]["imt"], level=task_arguments["disagg"]["target_level"]
+        )
+        self.model.hazard_config.set_parameter("general", "calculation_mode", "disaggregation")
+        self.model.hazard_config.set_parameter(
+            "disaggregation", "disagg_outputs", " ".join(task_arguments["disagg"]["disagg_outputs"])
+        )
+        if mag_bin_width := task_arguments["disagg"]["mag_bin_width"]:
+            self.model.hazard_config.set_parameter("disaggregation", "mag_bin_width", str(mag_bin_width))
+        if distance_bin_width := task_arguments["disagg"]["distance_bin_width"]:
+            self.model.hazard_config.set_parameter("disaggregation", "distance_bin_width", str(distance_bin_width))
+        if coordinate_bin_width := task_arguments["disagg"]["coordinate_bin_width"]:
+            self.model.hazard_config.set_parameter("disaggregation", "coordinate_bin_width", str(coordinate_bin_width))
+        if num_epsilon_bins := task_arguments["disagg"]["num_epsilon_bins"]:
+            self.model.hazard_config.set_parameter("disaggregation", "num_epsilon_bins", str(num_epsilon_bins))
+        if disagg_bin_edges := task_arguments["disagg"]["disagg_bin_edges"]:
+            self.model.hazard_config.set_parameter("disaggregation", "disagg_bin_edges", str(disagg_bin_edges))
+
+    @staticmethod
+    def get_disagg_description(task_arguments: Dict[str, Any]):
+        """get the description string for a disaggregation
+
+        Args:
+            task_arguments: a dict of task arguments
+        """
+        return (
+            f"Disaggregation for site: {task_arguments['site_params']['locations'][0]}, "
+            f"vs30: {task_arguments['site_params']['vs30']}, "
+            f"IMT: {task_arguments['hazard_curve']['imt']}, "
+            f"agg: {task_arguments['hazard_curve']['agg']}, "
+            f"{task_arguments['disagg']['poe']} in {task_arguments['disagg']['inv_time']} years"
+        )
+
     # TODO: we need to consider the best way to pass args to toshiAPI and make the args such as logic
```

Comment:
@chrisdicaprio let's brainstorm this one. Ideally we can retain flexibility and still meet these very desirable needs. Possibly a new API schema type / convention could suffice.